### PR TITLE
Enhance view for the electronic structure

### PR DESCRIPTION
**Pull request by superstar54**
This PR enhances the view of electronic structure.
User can 
- select dos_type (`atom` or `orbital`).
- change the energy range.
- plot or hide the fermi level line.

Here is a screenshot:

Screenshot from 2023-03-06 12-14-19

### CODE REVIEW

**Comment by superstar54** [comment]
As suggested by Giovanni, I removed the `energy_range` and `plot_fermi_level`.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @superstar54 for this change. I have some requests that better to improve and more discussion maybe needed.

File: aiidalab_qe/node_view.py [hunk 1]
```diff
@@ -140,12 +140,12 @@ def cmap(label: str) -> str:
 
 def _projections_curated(
     projections: ProjectionData,
-    curated_tag="atom",
+    dos_group="atom",
```
[the remark this answers is not in the document]
Comment (superstar54) [reply aiidalab_qe/node_view.py:143]: I changed it to `group_dos_by`

File: aiidalab_qe/node_view.py [hunk 2]
```diff
@@ -597,13 +604,57 @@ def _show_structure(self):
...
+                dos=dos_data,
+            )
+            self.result_tabs.children[2].children = [
+                settings,
+                _bands_plot_view,
+            ]
+
+        dos_group.observe(response, names="value")
+        #
+        self.result_tabs.children[2].children = [
```
[the remark this answers is not in the document]
Comment (superstar54) [reply aiidalab_qe/node_view.py:654]: I added a comment.
`# update the electronic structure tab`

File: aiidalab_qe/node_view.py [hunk 3]
```diff
@@ -597,13 +604,57 @@ def _show_structure(self):
...
-        self.result_tabs.children[2].children = [self._bands_plot_view]
+
+        def response(change):
+            data = export_data(self.node, dos_group=dos_group.value)
+            bands_data = data.get("bands", None)
+            dos_data = data.get("dos", None)
+            _bands_plot_view = BandsPlotWidget(
+                bands=bands_data,
+                dos=dos_data,
+            )
```
[the remark this answers is not in the document]
Comment (superstar54) [reply aiidalab_qe/node_view.py:649]: Thanks for the suggestion. I convert the data to the correct type and try:
```python
_bands_plot_view.bands = bands_data
_bands_plot_view.dos = dos_data
```
or using the original data and
```
_bands_plot_view.__init__(bands_data, dos_data)
```
 But the widget does not update in all the cases.

Better to discuss this with Dou.

**Comment by superstar54** [comment]
As discussed with Jusong, it is better to group the orbital by angular momentum (e.g. s, p, d) instead of both angular and magnetic momentum (e.g. px, py, pz). This is implemented.

Screenshot from 2023-03-15 10-02-19

File: aiidalab_qe/node_view.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (unkcpz) [reply aiidalab_qe/node_view.py:649]: Okay, then we don't worry about this for the moment.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @superstar54, just some minor things not very clear to me. Can you try running on a structure that has two atoms of the same elements in the cell, the Silicon from the example is the case. I am curious how this chang will parse it.

**Comment by unkcpz** [comment]
Hi @superstar54, are you able to manage to finish this today? I want to make a release for tomorrow's demo test.

File: aiidalab_qe/node_view.py [hunk 4]
```diff
@@ -157,13 +157,18 @@ def _projections_curated(
             orbital_data["angular_momentum"], orbital_data["magnetic_number"]
         ).lower()
 
-        if curated_tag == "atom":
-            curated_tag_var = atom_position
-        else:
+        if group_dos_by == "atom":
+            dos_group_name = atom_position
+        elif group_dos_by == "angular":
+            # by orbital label
```
[the remark this answers is not in the document]
Comment (superstar54) [reply aiidalab_qe/node_view.py:163]: group by:
- angular momentum, e.g. s, p, d
- -angular_magnetic momentum, e.g. px, py, pz

**Review by unkcpz — APPROVED** [review]

**Pull request merged**